lib/handlers/base/api-handler.ts

```diff
@@ -187,10 +195,12 @@ export abstract class APIGLambdaHandler<
               ({ body, statusCode } = handleRequestResult);
             }
           } catch (err) {
-            log.error({ err }, 'Unexpected error in handler');
             if (err instanceof CustomError) {
-              return err.toJSON(id);
+              const errorJson = err.toJSON(id);
+              log.error({ errorJson }, 'Unexpected error in handler');
+              return errorJson;
```

Comment:
is this fix suggesting that `return err.toJSON(id)` somehow is significantly more expensive than `const errJson = err.toJSON(id); return errJson;`?

Reply:
i just tried to maintain same behavior by logging the err.toJSON(id)